scripts/check-type-dependencies.js

```diff
@@ -87,7 +87,7 @@ function checkTypes(pkg) {
   );
   const deps = (typeDecl.match(/from '.*'/g) || [])
     .map(match => match.replace(/from '(.*)'/, '$1'))
-    .filter(n => !n.startsWith('.'));
+    .filter(n => !n.startsWith('.') && !n.startsWith(`${pkg.name}/`));
```

Comment:
this is because it started emitting code like

```ts
import { ApiRef as ApiRef$1, Observable as Observable$1 } from '@backstage/core-api';
import { ApiRef as ApiRef$2 } from '@backstage/core-api/src/apis/ApiRef';
```

Reply:
That code seems very broken though?

Reply:
Yep you are right, the replace actually is wrong, and i should have checked for equality OR starts with x+slash

Reply:
That too, but I mean the generated code :p

Reply:
Actually not sure if it's a problem.

Reply:
I mean, does typescript really care as such what the identifiers of things are? It just wants that the resulting shapes match so to speak

Reply:
The path `@backstage/core-api/src/apis/ApiRef` does not exist in published packages, so if the generated type definitions contain those paths, that'll lead to broken published packages.

Reply:
Ah right, good eye - yeah maybe that's an actual TS bug then.